clients/gitlabrepo/tarball.go

```diff
@@ -114,40 +122,85 @@ func (handler *tarballHandler) setup() error {
 func (handler *tarballHandler) getTarball() error {
 	url := fmt.Sprintf("%s/api/v4/projects/%d/repository/archive.tar.gz?sha=%s",
 		handler.repourl.Host(), handler.repo.ID, handler.commitSHA)
+
+	// Create a temp file.  This automatically appends a random number to the name.
+	tempDir, err := os.MkdirTemp("", repoDir)
+	if err != nil {
+		return fmt.Errorf("os.MkdirTemp: %w", err)
+	}
+	repoFile, err := os.CreateTemp(tempDir, repoFilename)
+	if err != nil {
+		return fmt.Errorf("%w io.Copy: %v", errTarballNotFound, err)
+	}
+	defer repoFile.Close()
+	err = handler.apiFunction(url, tempDir, repoFile)
+	if err != nil {
+		return fmt.Errorf("gitlab.apiFunction: %w", err)
+	}
+	// Gitlab url for pulling combined ci
+	url = fmt.Sprintf("%s/api/v4/projects/%d/ci/lint",
+		handler.repourl.Host(), handler.repo.ID)
+	ciFile, err := os.CreateTemp(tempDir, "gitlabscorecard_lint*.json")
+	if err != nil {
+		return fmt.Errorf("os.CreateTemp: %w", err)
+	}
+	err = handler.apiFunction(url, tempDir, ciFile)
+	if err != nil {
+		return fmt.Errorf("gitlab.apiFunction: %w", err)
+	}
+	byteValue, err := os.ReadFile(ciFile.Name())
+	if err != nil {
+		return fmt.Errorf("os.ReadFile: %w", err)
+	}
+	var result gitLabLint
+	err = json.Unmarshal(byteValue, &result)
+	if err != nil {
+		return fmt.Errorf("json.Unmarshal: %w", err)
+	}
+
+	ciYaml, err := os.Create(tempDir + "/gitlabscorecard_flattened_ci.yaml")
+	if err != nil {
+		return fmt.Errorf("os.CreateTemp: %w", err)
+	}
+	defer ciYaml.Close()
+	_, err = ciYaml.WriteString(result.MergedYaml)
+	if err != nil {
+		return fmt.Errorf("os.File.WriteString: %w", err)
+	}
+	err = ciYaml.Sync()
+	if err != nil {
+		return fmt.Errorf("os.File.Sync: %w", err)
+	}
+
+	handler.tempDir = tempDir
+	handler.tempTarFile = repoFile.Name()
+
+	handler.files = append(handler.files,
+		strings.TrimPrefix(ciYaml.Name(), filepath.Clean(handler.tempDir)+string(os.PathSeparator)))
+	return nil
+}
+
+func (handler *tarballHandler) apiFunction(url, tempDir string, repoFile *os.File) error {
 	req, err := http.NewRequestWithContext(handler.ctx, http.MethodGet, url, nil)
 	if err != nil {
-		return fmt.Errorf("http.NewRequestWithContext: %w", err)
+		return fmt.Errorf("%w io.Copy: %v", errTarballNotFound, err)
```

Comment:
Isn't this still an `http.NewRequestWithContext` error?

Reply:
Yep, I'll get that reverted... Refactoring, copy/paste issue.

Reply:
Just pushed the commit, reverting this. Thanks for the catch.